### PR TITLE
Validate Canadian Postal code

### PR DESCRIPTION
**Pull request by dsamojlenko**
# Summary | Résumé

Currently the Address form will accept anything up to 10 characters for a Postal Code. This just enforces proper postal code format.

To do this, the [Django localflavor package for Canada](https://django-localflavor.readthedocs.io/en/latest/localflavor/ca/) was added.

Note: we are not able to validate that the Postal code entered is the correct one for the Address, but we can at least ensure that the Postal code is correctly formatted and valid on its own.

### CODE REVIEW

File: register/forms.py [hunk 1]
```diff
@@ -107,7 +108,7 @@ class LocationAddressForm(HealthcareBaseForm, forms.Form):
     )
     city = forms.CharField(label=_("City"), max_length=100)
     province = forms.ChoiceField(label=_("Province or territory"), choices=provinces)
-    postal_code = forms.CharField(label=_("Postal code"), max_length=10)
+    postal_code = CAPostalCodeField(label=_("Postal code"), max_length=10)
```
Comment (dorianjp) [inline register/forms.py:111]: Do we want to add any hinting ahead, it looks like it's a hard regex on:
        'invalid': _('Enter a postal code in the format XXX XXX.'),

File: locale/fr/LC_MESSAGES/django.po [hunk 2]
```diff
@@ -2221,6 +2221,10 @@ msgstr "Province ou territoire"
 msgid "Postal code"
 msgstr "Code postal"
 
+#: register/forms.py
+msgid "Enter a valid Canadian postal code."
+msgstr ""
```
Comment (dsamojlenko) [inline locale/fr/LC_MESSAGES/django.po:2226]: @EliseKa

Comment (EliseKa) [reply locale/fr/LC_MESSAGES/django.po:2226]:
```suggestion
msgid "Enter a valid Canadian postal code."
msgstr "Entrez un code postal canadien valide."
```

File: register/forms.py (hunk 1 above)
Comment (dsamojlenko) [reply register/forms.py]: Got an updated validation error from Yedida

**Comment by dsamojlenko** [comment]
Updated the error message, ready for re-review

**Review by dorianjp — APPROVED** [review]
lgtm

**Pull request merged**